LSPS2/README.md

```diff
@@ -0,0 +1,864 @@
+
+* `option_zeroconf` is set.
+* `option_scid_alias` is set.
+* `announce_channel` is `false`.
+* The client-provided `to_self_delay` is less than or equal to
+  the agreed-upon `max_client_to_self_delay`.
+
+> **Rationale** `option_zeroconf` is set so that the payment can
+> be immediately forwarded to the client and resolved as soon as
+> the opening negotiation completes; if the client and LSP had to
+> wait for confirmation, the incoming payment would have to be
+> locked for an extended period, which is indistinguishable from
+> a channel jamming attack.
+> `option_zeroconf` cannot be set unless `announce_channel` is
+> `false`.
+> `option_scid_alias` needs to be set so that the channel can be
+> referred to on future invoices before the channel confirms.
+>
+> Future versions of this API may be able to utilize Asynchronous
+> Receive (currently being developed as of this version) to
+> be able to get around the `option_zeroconf` requirement, by
+> treating the receiver / client as offline until the client is
+> online *and* the JIT channel to it is deeply confirmed, if the
+> LSP and client negotiate a non-`option_zeroconf` channel for
+> the JIT channel.
+> A non-`option_zeroconf` JIT channel could also be later
+> published with `announce_channel = true`.
+
+Clients MAY reject the channel open attempt if any other
+parameters or options are not desired by the client (for example,
+it MAY reject non-zero channel reserve requirements imposed on
+the client by the LSP).
+Rejection is via the `error` Lightning BOLT message.
+
+LSPs MUST fail the payment with `unknown_next_peer` in case the
+client rejects the channel open via an `error`.
+
+If the client disconnects before the LSP can receive `funding_signed`
+from the client, the LSP MUST fail the incoming payment parts with
+`temporary_channel_failure`, then return to the previous step (waiting
+for payment).
+
+> **Rationale** The client might have crashed at this point, and
+> recovery might take an inordinate amount of time.
+> However, use of the `temporary_channel_failure` error informs the
+> payer that the error is temporary and it can retry later, if the
+> client is online at that point.
+
+Clients MUST NOT expect that `jit_channel_scid` is the same as
+the `alias` SCID.
+Clients MUST NOT expect that `jit_channel_scid` is different from
+the `alias` SCID.
+
+If the LSP does not support [LSPS3][] (Promise To Unconditionally Fund
+0-conf), then:
+
+* The LSP MUST immediately broadcast the funding transaction as soon
+  as it receives `funding_signed`.
+* The client MAY wait for the funding transaction to appear in its
+  mempool before it sends `channel_ready`.
+  * If it waits, it MUST validate that the funding transaction output
+    is expected (i.e. its amount and `scriptPubKey` are correct).
+
+If the LSP supports [LSPS3][], the client MAY use its `lsps3.getpromise`
+interface before it sends the `channel_ready` message.
+
+[LSPS3]: ../LSPS3/README.md
```

Comment:
Before we merge LSPS2, we need to have a group wide discussion around LSPS3. If we want to do it and what are the implications around it.